xbmc/cores/FFmpeg.cpp

```diff
@@ -178,7 +176,12 @@ std::tuple<uint8_t*, int> GetPacketExtradata(const AVPacket* pkt,
   if (ret < 0)
     return std::make_tuple(nullptr, 0);
 
-  bsf->par_in->codec_id = codecId;
+  ret = avcodec_parameters_copy(bsf->par_in, codecPar);
+  if (ret < 0)
+  {
+    av_bsf_free(&bsf);
```

Comment:
Looks like this could use a custom deleter to avoid catching all the av_bsf_free 😉

Reply:
Unfortunately we can not blindly wrap pointers to objects returned by ffmpeg because some ffmpeg functions (like `av_frame_free`) call `av_freep`. The only way to deal with non-smartpointer-aware C/C++ libraries is carefully think what you do and where :(

Reply:
In cases where the lifetime of an object is limited to one single function, we can implement smart pointer just fine. But in #21248 I tried to make `extract_extradata` returning unique_ptr and got immediate premature free.